internal/queue/consumer.go

```diff
@@ -1,46 +1,76 @@
 package queue
 
-import "errors"
+import (
+	"errors"
+	"sublinks/sublinks-federation/internal/worker"
+)
 
-func (q *RabbitQueue) createConsumer(queueName string) error {
+func (q *RabbitQueue) createConsumer(queueData ConsumerQueue) error {
 	channelRabbitMQ, err := q.Connection.Channel()
 	if err != nil {
 		return err
 	}
-	err = q.createQueue(channelRabbitMQ, queueName)
+	err = q.createQueue(channelRabbitMQ, queueData.QueueName)
 	if err != nil {
 		return err
 	}
+	err = channelRabbitMQ.QueueBind(
+		queueData.QueueName,  // queue name
+		queueData.RoutingKey, // routing key
+		queueData.Exchange,   // exchange
+		false,
+		nil)
+	if err != nil {
+		return err
+	}
+
 	// Subscribing to QueueService1 for getting messages.
 	messages, err := channelRabbitMQ.Consume(
-		queueName, // queue name
-		"",        // consumer
-		true,      // auto-ack
-		false,     // exclusive
-		false,     // no local
-		false,     // no wait
-		nil,       // arguments
+		queueData.QueueName, // queue name
+		"",                  // consumer
```

Comment:
When this is blank, it gets auto-assigned. _Might_ be ok? But might also be a better idea for us to think this through some more

Reply:
Maybe. Like I said. Need to think it through. Probably, should create a separate issue to track it after this PR gets merged